zerver/lib/error_notify.py

```diff
@@ -68,63 +42,47 @@ def zulip_server_error(report: Dict[str, Any], error_bot_email: str) -> None:
             if field == "QUERY_STRING":
                 val = clean_data_from_query_parameters(str(val))
             request_repr += f'- {field}: "{val}"\n'
-        request_repr += "~~~~"
     else:
-        request_repr = "Request info: none"
+        request_repr = "Request info: none\n"
 
-    message = f"""{logger_str}
+    if channel == "email":
+        message = f"""\
+{logger_str}
 Error generated by {user_info}
 
-~~~~ pytb
 {report['stack_trace']}
 
-~~~~
 {deployment}
-{request_repr}"""
-
-    error_bot_realm = get_realm(settings.STAFF_SUBDOMAIN)
-    error_bot = get_system_bot(error_bot_email, error_bot_realm.id)
-    errors_stream = get_stream("errors", error_bot_realm)
 
-    internal_send_stream_message(
-        error_bot,
-        errors_stream,
-        format_email_subject(email_subject),
-        message,
-    )
-
-
-def email_server_error(report: Dict[str, Any]) -> None:
-    email_subject = "{node}: {message}".format(**report)
-
-    logger_str = logger_repr(report)
-    user_info = user_info_str(report)
-    deployment = deployment_repr(report)
+{request_repr}"""
 
-    if report["has_request"]:
-        request_repr = """\
-Request info:
-- path: {path}
-- {method}: {data}
-""".format(
-            **report
-        )
-        for field in ["REMOTE_ADDR", "QUERY_STRING", "SERVER_NAME"]:
-            val = report.get(field.lower())
-            if field == "QUERY_STRING":
-                val = clean_data_from_query_parameters(str(val))
-            request_repr += f'- {field}: "{val}"\n'
+        mail_admins(topic, message, fail_silently=True)
     else:
-        request_repr = "Request info: none\n"
-
-    message = f"""\
-{logger_str}
+        assert settings.ERROR_BOT is not None
+        message = f"""{logger_str}
 Error generated by {user_info}
 
+~~~~ pytb
 {report['stack_trace']}
 
+~~~~
 {deployment}
-
 {request_repr}"""
 
-    mail_admins(format_email_subject(email_subject), message, fail_silently=True)
+        error_bot_realm = get_realm(settings.STAFF_SUBDOMAIN)
+        error_bot = get_system_bot(settings.ERROR_BOT, error_bot_realm.id)
+        errors_stream = get_stream("errors", error_bot_realm)
+
+        internal_send_stream_message(
+            error_bot,
+            errors_stream,
+            topic,
+            message,
+        )
+
+
+def do_report_error(report: Dict[str, Any]) -> None:
+    report = defaultdict(lambda: None, report)
+    send_error("email", report)
+    if settings.ERROR_BOT:
+        send_error("zulip", report)
```

Comment:
Should we consider removing the `ERROR_BOT` feature here as well? I don't think I've used that in recent years either, and I'm not sure I see it having any advantage over the "email" and "sentry" options we would have available with it removed.

(This could be a follow-up, if we want to remove the bot user too; @mateuszmandera thoughts?).

Reply:
I'd be happy to do that, as long as you think the email notifications are reasonable for self-hosters who don't set up Sentry.  Having them "in-app" is potentially somewhat nice -- but I agree that email is more straightforward than the magic `#errors` stream.

Reply:
Added a final commit which does that removal.

Reply:
Yeah I've always been a bit uncomfortable with the existence of the "send a zulip" feature since you can get an infinite error notification loop with that class of thing -- say a user in the destination realm has a browser client 

Email seems fine as a Sentry alternative; if we wanted one other thing, the thing I'd pick is probably "Write to a log file", not "Send a Zulip". But I think we can just not have it and see if anyone complains.